chips/rp2040/src/chip.rs

```diff
@@ -147,9 +149,14 @@ impl Rp2040DefaultPeripherals<'_> {
             pins: RPPins::new(),
             uart0: Uart::new_uart0(),
             adc: adc::Adc::new(),
+            spi0: spi::Spi::new_spi0(),
             sysinfo: sysinfo::SysInfo::new(),
         }
     }
+
+    pub fn set_clocks(&'a self) {
+        self.spi0.set_clocks(&self.clocks);
+    }
 }
```

Comment:
Wouldn't it be easier to just pass `clocks` in as an argument to `spi::new()`?

Reply:
I don't think so, it is a circular dependency, as peripherals has  the clock field. I can try though.

Reply:
Yeah, that won't work. I used a `setup_circular_dependencies()` method to address that issue for some other chips, which at least made it only require a single call in the case where there were several circular dependencies.

Reply:
I followed your code 😃